packages/fuels-core/src/code_gen/functions_gen.rs

```diff
@@ -264,28 +244,32 @@ mod tests {
             .collect::<HashMap<usize, TypeDeclaration>>();
 
         // Grabbing the one and only function in it.
-        let result = expand_function(&parsed_abi.functions[0], &all_types);
-
-        // let result = expand_function(&the_function, &Default::default(), &Default::default());
-        let expected = TokenStream::from_str(
-            r#"
-            #[doc = "Calls the contract's `some_abi_funct` (0x00000000652399f3) function"]
-            pub fn some_abi_funct(&self, s_1: MyStruct1, s_2: MyStruct2) -> ContractCallHandler<MyStruct1> {
-                let provider = self.wallet.get_provider().expect("Provider not set up");
-                Contract::method_hash(
-                    &provider,
-                    self.contract_id.clone(),
-                    &self.wallet,
-                    [0, 0, 0, 0, 101 , 35 , 153 , 243],
-                    &[s_1.into_token(), s_2.into_token(),]
-                ).expect("method not found (this should never happen)")
-            }
-
-            "#,
-        );
-        let expected = expected?.to_string();
-
-        assert_eq!(result?.to_string(), expected);
+        let result = expand_function(&parsed_abi.functions[0], &all_types)?;
+
+        let expected_code = r#"
+                #[doc = "Calls the contract's `some_abi_funct` function"]
+                pub fn some_abi_funct(&self, s_1: MyStruct1, s_2: MyStruct2) -> ContractCallHandler<MyStruct1> {
+                    let provider = self.wallet.get_provider().expect("Provider not set up");
+                    let encoded_fn_selector = resolve_fn_selector(
+                        "some_abi_funct",
+                        &[<MyStruct1> :: param_type(), <MyStruct2> :: param_type()]
+                    );
```

Comment:
Ya love to see it! 🤘 

I really disliked having to maintain that hardcoded selector byte array there.